simulator-spring-boot/src/main/java/org/citrusframework/simulator/http/HttpRequestPathScenarioMapper.java

```diff
@@ -103,4 +112,35 @@ public void setScenarioList(List<SimulatorScenario> scenarioList) {
     public void setConfiguration(SimulatorConfigurationProperties configuration) {
         this.setSimulatorConfigurationProperties(configuration);
     }
+
+    @Override
+    public void setApplicationContext(@NonNull ApplicationContext applicationContext)
+        throws BeansException {
+        this.applicationContext = applicationContext;
+    }
+
+    @Override
+    public void onApplicationEvent(@NonNull ScenariosReloadedEvent event) {
+        setScenariosFromApplicationContext();
+    }
+
+    private void setScenariosFromApplicationContext() {
+        updateScenarioList(
+            filterScenarios(applicationContext.getBeansOfType(SimulatorScenario.class).values()));
+    }
+
+    private List<HttpScenario> filterScenarios(Collection<SimulatorScenario> allScenarios) {
+        return allScenarios.stream().filter(HttpScenario.class::isInstance)
+            .map(HttpScenario.class::cast)
+            .sorted(new HttpPathSpecificityComparator())
```

Comment:
couldn't this be a singleton? why is it stateful? I think the initialization per request could have performance impact. correct me if I am wrong!